src/binder/bind/bind_graph_pattern.cpp

```diff
@@ -434,12 +436,12 @@ void Binder::bindQueryNodeProperties(NodeExpression& node) {
 
 std::vector<common::table_id_t> Binder::bindNodeTableIDs(
     const std::vector<std::string>& tableNames) {
-    if (!catalog.getReadOnlyVersion()->hasNodeTable()) {
+    if (catalog.getReadOnlyVersion()->getAllNodeTableIDs().empty()) {
```

Comment:
Up to u, but I think the previous function is more clear.

Reply:
hasNodeTable() is only used here, so it's unnecessary to keep it in the catalog.